### PR TITLE
Tfidf

### PR DESCRIPTION
**Pull request by danielsgriffin**
- Added class TF_IDFSummarizer to sentence_summarizer.
- Modified gap_fill_generator to allow an option of summarizer.
- Modified several articles to remove Photo Credits that appeared from haphazard copy-pasting.

### CODE REVIEW

File: gadfly/gap_fill_generator.py [hunk 1]
```diff
@@ -30,15 +30,28 @@ def find_named_entities(self):
         return entities
 
     def summarize_sentences(self):
```
Comment (vijayv) [inline gadfly/gap_fill_generator.py]: Can you refactor this to follow the pattern that I used for the blank_types? Look at the generate_questions method in the development branch now.

File: gadfly/sentence_summarizer.py [hunk 2]
```diff
@@ -58,3 +59,78 @@ def summarize(self, sents, n):
...
         return nlargest(n, ranking, key=ranking.get)
+
+
+class TF_IDFSummarizer:
+    def __init__(self):
+        pass
+
+    def get_tokens_and_freqs(self, sents, style):
+        """Get a list of all tokens in sents and a Counter of them"""
+        if style == 'standard':
```
Comment (vijayv) [inline gadfly/sentence_summarizer.py]: Can you change this to an ENUM? See the pattern used in gap_fill_generator  in development branch.

File: gadfly/sentence_summarizer.py [hunk 3]
```diff
@@ -58,3 +59,78 @@ def summarize(self, sents, n):
...
+        document_freq_dict = Counter(all_tokens)
+        return document_freq_dict, all_tokens
+
+    def get_dict_tf_idf(self, sents, document_freq_dict, all_tokens, style):
+        """Get a dict of _____"""
+        dict_tf_idf = {}
+        for word in set(all_tokens):
+            for n, sent in enumerate(sents):
+                sum = 0
+                if style == 'lower':
```
Comment (vijayv) [inline gadfly/sentence_summarizer.py]: Use ENUM

**Comment by vijayv** [comment]
@danielsgriffin Looks pretty good. Can you 

1. merge development into this branch and resolve the conflicts
2. Use Enums in the places specified (see inline comments)

thanks!

**Comment by danielsgriffin** [comment]
new pushes

**Pull request closed without merging**